### ISSUE TITLE
Xamarin iOS OCR 

### ISSUE THREAD

**Issue by ziziana** [doo/scanbot-sdk-example-xamarin#1]
Hi,

I am trying to use this library to read text from an image using OCR in Xamarin iOS language. I couldn't find any example for Xamarin iOS, so I tried to figure out by myself how to do this and tried this code:

` NSUrl url = new NSUrl("http://digitalnativestudios.com/textmeshpro/docs/rich-text/line-indent.png");
           var rect = CGRect.Infinite;
            SBSDKOpticalTextRecognizer.RecognizeText(url, rect, "en", HandleSBSDKCompletionHandler);
`

But, I get the following error: 

> Scanbot SDK: ERROR - Language data not found for all specified languages

Could you help me with an example of how can I use OCR for Xamarin iOS?

Thank you

**Comment by tailsu** [comment]
Hi Ziziana, I'll prepare an example tomorrow, but the gist is that you also need to add the OCR bundle into your project. Check out how our Xamarin.Forms example app for iOS is configured: https://github.com/doo/scanbot-sdk-example-xamarin-forms/tree/master/iOS

**Comment by tailsu** [comment]
I've pushed an example how to use the OCR functionality. The important step is adding the SBSDKLanguageData folder to your solution with all the necessary .traineddata files in it marked as `BundleResource`. Check out the Resources/ folder in the project. As for the OCR itself we provide a convenience wrapper in the SBSDK class that's easier to use than the iOS SDK classes.

Check it out:

https://github.com/doo/scanbot-sdk-example-xamarin/blob/master/iOS/ViewControllers/MainSelectionTableViewController.cs#L111

**Comment by tailsu** [comment]
I forgot to mention - it's possible to download the traineddata files at runtime, instead of bundling them into the app package, even though the demo doesn't show how to do this. I can add a demo of that as well should you find the need for it.

**Comment by ziziana** [comment]
@tailsu About the language bundling it's ok for now. Thank you very much for your help, your example saved me some time.

**Comment by ziziana** [comment]
@tailsu I took a picture to this image from Google (a boarding pass) https://png.pngtree.com/element_origin_min_pic/17/01/04/f4ab0ce6a95f9a6a17dc93a5adb8bdda.jpg 
and the code prints this text from the image: .uuhunuﬂ. .‘Ilnmxlxz , l. 5::
mhﬁ‘ixwﬁﬂﬁiéﬂ

gumltit i‘vhwmﬂﬁzlxxli.
\‘ﬁﬂhﬁhﬁi

Should I use some other functions from SDK in order to receive a normal text?

Thanks!

**Comment by tailsu** [comment]
The resolution of the letters on this image is below the OCR engine's threshold. The engine is best suited to work with scanned documents or documents captured with the camera, where the resolution is much higher.

**Comment by ziziana** [comment]
I see. I asked this question because with Tesseract it worked a bit better.
It managed to show some letters from the picture and I thought that maybe
there's something more to do when using scanbot.

**Comment by tailsu** [comment]
Nope, there isn't. Do you really need the OCR engine to work with such low-res images?

**Comment by ziziana** [comment]
No, the examples I gave you were just for test. I will test more tomorrow
on real documents and see how it works. I just wanted to make sure that I
don't have to add other code. Thank you very much for your help :)
